### PR TITLE
[new] Add Infusion@2.0.0 w/ npm auto-update

### PR DESCRIPTION
**Pull request by waharnum**
This PR adds the latest version of Infusion, a Javascript framework used by a number of international research teams working on accessible software projects and in a number of other educational contexts.

I'm one of the contributors and suggested adding it to cdnjs in this thread on our work mailing list: http://lists.idrc.ocad.ca/pipermail/fluid-work/2017-January/010222.html

---

# Profile of the lib
 * Git repository (required): http://github.com/fluid-project/infusion.git
 * Official website (optional, not the repository): http://www.fluidproject.org/
 * NPM package url (optional): https://www.npmjs.com/package/infusion
 * License and its reference: BSD / ECL 2.0, https://github.com/fluid-project/infusion/blob/master/Infusion-LICENSE.txt
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 81
   - Count of watchers: 16
   - Count of forks: 72
 * NPM download stats (optional):
   - Downloads in the last day: 24
   - Downloads in the last week: 361
   - Downloads in the last month: 2066

# Essential checklist
 * [x] I'm the author of this library (contributor)
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/infusion/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  "name": "infusion",
+  "description": "Infusion is an application framework for developing flexible stuff with JavaScript",
+  "version": "2.0.0",
+  "filename": "infusion-all.min.js",
+  "author": "Fluid Project",
+  "homepage": "http://www.fluidproject.org/",
+  "dependencies": {
+    "fluid-resolve": "1.2.0"
+  },
+  "license": "(BSD-3-Clause OR ECL-2.0)",
```
Comment (pvnr0082t) [inline ajax/libs/infusion/package.json]: Let's use licenses array as following:
```json
  "licenses": [
    "BSD-3-Clause",
    "ECL-2.0"
  ],
```
Thanks

**Comment by waharnum** [comment]
@pvnr0082t - by source files, do you mean the unminified consolidated files, or the individual source files that are available at https://github.com/fluid-project/infusion/tree/master/src?

**Comment by pvnr0082t** [comment]
@waharnum I meant the former. Sorry for confusing you.

**Comment by waharnum** [comment]
@pvnr0082t minified CSS added - we don't currently minify JS/CSS for our /dist directory in NPM, but I will work on making that a part of our publish process to NPM for the next version.

**Comment by pvnr0082t** [comment]
@waharnum Okay. Nice work!

**Comment by waharnum** [comment]
@pvnr0082t thank you for speedy review - I'm excited to be able to put Infusion on cdnjs!

**Comment by waharnum** [comment]
Hello - checking in whether there is further work needed from our end to move this forward? I believe I've addressed all @pvnr0082t's comments. It would be great to have 2.0.0 release available

**Comment by pvnr0082t** [comment]
ping @kennynaoh

**Comment by maruilian11** [comment]
Hi @waharnum, could you please also remove `dependencies` since we don't need it. Thanks!

**Comment by waharnum** [comment]
Hi @maruilian11 - I've removed the `dependencies` block in `package.json`.

Regarding file permissions, I don't see any files that have permissions other than `-rw-r--r--` (at least on my local checkout) except for the directories, which have the standard `drwxr-xr-x` permission structure. Is there a specific file or files that are an issue? Sorry if I'm missing something obvious!

**Comment by maruilian11** [comment]
@waharnum sorry, I think the permission is ok. could you rebase this PR with our latest master by using `git rebase master`. Thanks!

**Comment by PeterDaveHello** [comment]
@maruilian11 You should confirm that the contributor is using a GitHub gui or a local command line first before asking rebase ... btw, for non-active users without willing to clone the repo for just rebase, I think you should try to help rebase (rebase by you) even with a squash/fixup process.

**Comment by maruilian11** [comment]
@PeterDaveHello Got it.
@waharnum I help you rebase this branch with the latest master branch and squash the commits, I hope you won't mind.

**Comment by PeterDaveHello** [comment]
@maruilian11 your sentence looks strange ...

**Comment by maruilian11** [comment]
LGTM

**Comment by maruilian11** [comment]
@PeterDaveHello hmm... you mean the sentence I said to waharnum?

**Comment by PeterDaveHello** [comment]
@maruilian11 check it yourself ...

**Comment by maruilian11** [comment]
@PeterDaveHello Just modify it. Could you help me check this PR?

**Comment by PeterDaveHello** [comment]
@maruilian11 I just don't get it, why don't you ask the contributor before the rebasing? I don't believe that is how the open source community works.

**Comment by maruilian11** [comment]
@PeterDaveHello hmm... Because I think the PR only need to be rebased, I thought it would be help....
I will ask the contributor first next time... sorry...

**Comment by waharnum** [comment]
No problem on the rebase, thank you @maruilian11. Anything further needed on this one to make the PR acceptable? Thanks again!

**Comment by PeterDaveHello** [comment]
Looks like the most versions on npm package are not for production but development builds ... so should update `.gitignore` to ignore them.
 
 - 2.0.0-dev.20151007T134312Z.8c33cd4
 - 2.0.0-dev.20151024T031903Z.3da18aa
 - 2.0.0-dev.20151027T140548Z.c9d974c
 - 2.0.0-dev.20151103T162114Z.69aa6c8
 - 2.0.0-dev.20151103T223642Z.130b29c
 - 2.0.0-dev.20151113T224751Z.3ba00f2
 - 2.0.0-dev.20151123T190910Z.35b21fd
 - 2.0.0-dev.20151130T180545Z.18ee3f8
 - 2.0.0-dev.20151211T025229Z.eabb4fc
 - 2.0.0-dev.20160115T165518Z.2f38c09
 - 2.0.0-dev.20160309T145652Z.9f515ec
 - 2.0.0-dev.20160505T142427Z.63b8a53
 - 2.0.0-dev.20160513T002747Z.477b81a
 - 2.0.0-dev.20160519T222603Z.754d2c6
 - 2.0.0-dev.20160902T155354Z.b963420
 - 2.0.0-dev.20160911T222604Z.b86ecbb
 - 2.0.0-dev.20161002T002156Z.a2802df
 - 2.0.0-dev.20161027T155641Z.0bcd507
 - 2.0.0-dev.20161124T191742Z.17a345b
 - 2.0.0-dev.20161219T154515Z.0f2ccc8
 - 2.0.0-dev.20161219T170555Z.5778f7e
 - 3.0.0-dev.20170118T192406Z.368093e
 - 3.0.0-dev.20170127T130413Z.103de6e
 - 3.0.0-dev.20170128T013435Z.82e259f
 - 3.0.0-dev.20170131T153243Z.6aab53a

**Comment by waharnum** [comment]
@PeterDaveHello - yes, the nature of our work means dev packages are published to NPM so that they can be sourced from other work. I was going to add a blanket `*-dev.*` ignore but see you have already done so. Thanks!

I have an outstanding PR to generate minified `/dist` files automatically for our NPM publish process that will be in before our next major release.

Thank you for the merge!

**Comment by PeterDaveHello** [comment]
@waharnum no problem, thanks for your contribution.

**Pull request merged**